File: population/dashboards/population.py
from pymongo import MongoClient
from config.config import Config  # ✅ Import class Config
import pandas as pd

# Kết nối MongoDB
client = MongoClient(Config.MONGO_URI)
db = client[Config.DATABASE_NAME]
population_collection = db[Config.POPULATION_COLLECTION]  # Lấy collection

# Lấy dữ liệu dân số từ MongoDB


def get_population_data():
    cursor = population_collection.find({}, {"_id": 0})  # ✅ Truy vấn đúng cách
    df = pd.DataFrame(list(cursor))

    if df.empty:
        print("⚠️ Không có dữ liệu dân số trong MongoDB!")
        return pd.DataFrame()

    df.rename(columns={
        'province': 'province',
        'year': 'year',
        'population_density': 'populationDensity',
        'average_population': 'averagePopulation',
        'gender_ratio': 'genderRatio',
        'population_growth_rate': 'populationGrowthRate',
        'labor_force': 'laborForce',
        'region': 'region'
    }, inplace=True)

    return df

File: population/backend/models/Population.js
const mongoose = require("mongoose");
const ProvinceSchema = new mongoose.Schema(
  {
    province: { type: String, required: true }, // Tỉnh thành
    populationDensity: { type: Number, required: true }, // populationDensity (người/km²)
    averagePopulation: { type: Number, required: true }, // averagePopulation (nghìn người)
    genderRatio: { type: Number, required: true }, // genderRatio (nữ/100 nam)
    populationGrowthRate: { type: Number, required: true }, // populationGrowthRate (%)
    laborForce: { type: Number, required: true }, // Lực lượng lao động từ 15 tuổi trở lên (nghìn người)
    region: { type: String, required: true }, // region địa lý
    year: { type: Number, required: true }, // Năm thống kê
  },
  { collection: "population" }
);

const Province = mongoose.model("Province", ProvinceSchema);

module.exports = Province;
